herddb-core/src/main/java/herddb/data/consistency/TableDataChecksum.java

```diff
@@ -0,0 +1,113 @@
+ distributed with this work for additional information
+ regarding copyright ownership. Diennea S.r.l. licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+
+ */
+package herddb.data.consistency;
+
+import herddb.codec.RecordSerializer;
+import herddb.core.TableSpaceManager;
+import herddb.model.TransactionContext;
+import herddb.model.DataScanner;
+import herddb.model.DataScannerException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import herddb.model.commands.ScanStatement;
+import herddb.utils.DataAccessor;
+import herddb.core.AbstractTableManager;
+import herddb.core.DBManager;
+import herddb.model.Column;
+import herddb.model.Table;
+import herddb.sql.TranslatedQuery;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import net.jpountz.xxhash.StreamingXXHash64;
+import net.jpountz.xxhash.XXHashFactory;
+
+/**
+ * digest creation by scanning the table
+ *
+ * @author Hamado.Dene
+ */
+public abstract class TableDataChecksum {
+
+    private static final Logger LOGGER = Logger.getLogger(TableDataChecksum.class.getName());
+    private static final XXHashFactory factory = XXHashFactory.fastestInstance();
+    //private static final int SEED = 0x9747b28c;
+    private static final int SEED = 0;
+    public static final boolean DIGEST_NOT_AVAILABLE = true;
+    public static final String HASH_TYPE = "StreamingXXHash64";
+    public static int NRECORDS = 0;
+    public static long SCAN_DURATION = 0;
+    private static TranslatedQuery translated;
+
+    public static TableChecksum createChecksum(DBManager manager, TranslatedQuery query, TableSpaceManager tableSpaceManager, String tableSpace, String tableName) throws DataScannerException {
+
+        AbstractTableManager tablemanager = tableSpaceManager.getTableManager(tableName);
+        String nodeID = tableSpaceManager.getDbmanager().getNodeId();
+        if (query == null) {
+            final Table table = tableSpaceManager.getTableManager(tableName).getTable();
+            String columns = parseColumns(table);
+            translated = manager.getPlanner().translate(tableSpace, "SELECT  "
+                    + columns
+                    + " FROM " + tableName
+                    + " order by "
+                    + parsePrimaryKeys(table), Collections.emptyList(), true, false, false, -1);
+        }
+        translated = query;
```

Comment:
why do you need a static variable ? can't you simply use a variable inside this method ?